src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc

```diff
@@ -109,103 +127,461 @@ class PerformAudioPassThruRequestTest
+    ON_CALL(hmi_interfaces_, GetInterfaceFromFunction(_))
+        .WillByDefault(Return(am::HmiInterfaces::HMI_INTERFACE_UI));
+    DefineUIsAvailable();
+  }
+
   sync_primitives::Lock lock_;
   NiceMock<MockHmiInterfaces> hmi_interfaces_;
   MockMessageHelper& mock_message_helper_;
   MockAppPtr mock_app_;
 };
 
 TEST_F(PerformAudioPassThruRequestTest, OnTimeout_GENERIC_ERROR) {
-  MessageSharedPtr msg_ui = CreateMessage(smart_objects::SmartType_Map);
-  (*msg_ui)[am::strings::msg_params][am::strings::result_code] =
+  MessageSharedPtr msg_mobile_response =
+      CreateMessage(smart_objects::SmartType_Map);
+  (*msg_mobile_response)[am::strings::msg_params][am::strings::result_code] =
       am::mobile_api::Result::GENERIC_ERROR;
-  (*msg_ui)[am::strings::msg_params][am::strings::success] = false;
+  (*msg_mobile_response)[am::strings::msg_params][am::strings::success] = false;
 
   utils::SharedPtr<PerformAudioPassThruRequest> command =
       CreateCommand<PerformAudioPassThruRequest>();
 
   EXPECT_CALL(app_mngr_, EndAudioPassThrough()).WillOnce(Return(true));
   EXPECT_CALL(app_mngr_, StopAudioPassThru(_));
-
   EXPECT_CALL(
       mock_message_helper_,
       CreateNegativeResponse(_, _, _, am::mobile_api::Result::GENERIC_ERROR))
-      .WillOnce(Return(msg_ui));
-
-  MessageSharedPtr vr_command_result;
+      .WillOnce(Return(msg_mobile_response));
   EXPECT_CALL(
       app_mngr_,
       ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
-      .WillOnce(DoAll(SaveArg<0>(&vr_command_result), Return(true)));
+      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
 
   command->onTimeOut();
-  EXPECT_EQ((*vr_command_result)[am::strings::msg_params][am::strings::success]
-                .asBool(),
-            false);
-  EXPECT_EQ(
-      (*vr_command_result)[am::strings::msg_params][am::strings::result_code]
-          .asInt(),
-      static_cast<int32_t>(am::mobile_api::Result::GENERIC_ERROR));
-  Mock::VerifyAndClearExpectations(&mock_message_helper_);
+
+  ResultCommandExpectations(msg_mobile_response,
+                            kEmptyInfo,
+                            am::mobile_api::Result::GENERIC_ERROR,
+                            false);
 }
 
 TEST_F(PerformAudioPassThruRequestTest,
        OnEvent_UIHmiSendUnsupportedResource_UNSUPPORTED_RESOURCE) {
-  MessageSharedPtr msg_ui = CreateFullParamsUISO();
-  (*msg_ui)[am::strings::params][am::strings::connection_key] = kConnectionKey;
-
+  MessageSharedPtr msg_mobile = CreateMobileMessageSO();
   utils::SharedPtr<PerformAudioPassThruRequest> command =
-      CreateCommand<PerformAudioPassThruRequest>(msg_ui);
+      CreateCommand<PerformAudioPassThruRequest>(msg_mobile);
+
+  MessageSharedPtr msg_tts =
+      CreateHMIMessageSO(hmi_apis::Common_Result::SUCCESS, kCmdId);
+  am::event_engine::Event event_tts(hmi_apis::FunctionID::TTS_Speak);
+  event_tts.set_smart_object(*msg_tts);
```

Comment:
@tkireva I see a lot of code with creation event. Please create separate method for creation event. I think it will be better.